Automattic/wp-calypso
# Pricing page rework: Display additional legacy product cards

This PR updates the product and bundle lists to display additional legacy product card(s) if they are currently owned/active on the site (for a signed-in user)

#### Proposed Changes

* Add icons for Security and Complete bundles
* Move `getCtaLabel` to `useStoreItemInfo` hook for reusability
* Show "All bundles" section when a legacy item is owned by a user
* Fix product name i18n when name is JSX Element

#### Testing Instructions

- Create a JN site and connect Jetpack
- Purchase a legacy product for the site e.g. Security Daily using this link after replacing `:site` with the actual site URL `https://wordpress.com/checkout/:site/jetpack_security_daily`
- Do any one of these
    - Open Calypo Blue and Green in two separate tabs
        - Click on Jetpack Cloud Live link below and wait for the redirect to complete, then goto `/pricing/:site?flags=jetpack/pricing-page-rework-v1`
        - Click on Calypso Live link below and wait for the redirect to complete, then goto `/plans/:site?flags=jetpack/pricing-page-rework-v1`
    - or boot up this PR 
        - Run `git fetch && git checkout fix/jp-pricing-rework/legacy-products`
        - Run `yarn start` in one terminal window and `yarn start-jetpack-cloud-p` in another one.
        - Open this in one tab `http://calypso.localhost:3000/plans/:site?flags=jetpack/pricing-page-rework-v1`
        - Open this in another one `http://jetpack.cloud.localhost:3001/pricing/:site?flags=jetpack/pricing-page-rework-v1&view=bundles`
- Confirm these steps for both Calypso Blue and Green
    - Confirm that you see the new pricing UI
    - Confirm that you see the "Backup Daily" card in "All products" section
    - Goto Bundles tab
    - Confirm that you see "All bundles" section
    - Confirm that you see "Security Daily" card in the list
- Goto `/pricing/?flags=jetpack/pricing-page-rework-v1` for Jetpack Cloud
- Confirm that you don't see "All bundles" section

#### Pre-merge Checklist






Cpmpletes: 1202796695664022-as-1202924222500997/f

<img width="1012" alt="Screenshot 2022-09-06 at 3 36 01 PM" src="https://user-images.githubusercontent.com/18226415/188645291-e756b5c5-3050-4350-b147-2dd20644632f.png">
<img width="1223" alt="Screenshot 2022-09-06 at 6 29 06 PM" src="https://user-images.githubusercontent.com/18226415/188645302-a408c73a-eb17-4b97-ae5a-63eb60f5110b.png">
<img width="1368" alt="Screenshot 2022-09-06 at 6 43 07 PM" src="https://user-images.githubusercontent.com/18226415/188645310-ea5d2b29-c97e-4481-bb5d-244b3b0fd10d.png">
<img width="1373" alt="Screenshot 2022-09-06 at 6 43 22 PM" src="https://user-images.githubusercontent.com/18226415/188645317-68295e4c-2cfb-4151-9cac-a33e22f3fcd3.png">
<img width="1218" alt="Screenshot 2022-09-06 at 6 43 45 PM" src="https://user-images.githubusercontent.com/18226415/188645323-ba372ab6-c2a6-42ca-8991-757bff39a7f6.png">


---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202924222500997
